libs/flag.js

```diff
@@ -148,3 +150,51 @@ exports.unflag = function (aModel, aContent, aUser, aCallback) {
     }
   });
 };
+
+exports.setFlaggedListInModel = function (aModelName, aOptions, aCallback) {
```

Comment:
This is definitely the wrong module for this function. This module is supposed to be completely devoid of any front-end view logic (notice how "aOptions" or "options" are mentioned nowhere else in this file). This module defines the logic for how flagging is handled.